src/System.Management.Automation/engine/remoting/commands/PushRunspaceCommand.cs

```diff
@@ -1272,10 +1284,12 @@ private RemoteRunspace GetRunspaceForSSHSession()
         {
             var sshConnectionInfo = new SSHConnectionInfo(this.UserName, ResolveComputerName(HostName), this.KeyFilePath, this.Port);
             var typeTable = TypeTable.LoadDefaultTypeFiles();
-            var remoteRunspace = RunspaceFactory.CreateRunspace(sshConnectionInfo, this.Host, typeTable) as RemoteRunspace;
-            remoteRunspace.Open();
-            remoteRunspace.ShouldCloseOnPop = true;
-
+            _tempRunspace = RunspaceFactory.CreateRunspace(sshConnectionInfo, this.Host, typeTable) as RemoteRunspace;
```

Comment:
You should comment why _tempRunspace is being used.  On the surface, the usage either implies reentrancy or threading issue; if so, StopProcessing's use of _tempRunspace is problematic.  Comments explaining it's expected usage would help.

Reply:
This is a common pattern we use for Cmdlet async StopProcessing.  I'll add a comment.

Reply:
Regarding above comment about connection established.  I think it is possible for an error to occur before the first PSRP message is processed.

Reply:
However, we want to do immediate connection clean up only if PSRP messages are being processed because then we can rely on the protocol to clean up the client connection state.  This is just for the case where the session is abruptly severed and we can't count on PSRP for clean up.  I'll add comment to clarify.